Xamarin.Forms.Platform.Android/Renderers/ShellToolbarTracker.cs

```diff
@@ -235,51 +236,114 @@ protected virtual async void UpdateLeftBarButtonItem(Context context, Toolbar to
 			var backButtonHandler = Shell.GetBackButtonBehavior(page);
 			toolbar.SetNavigationOnClickListener(this);
 
+			var activity = (FormsAppCompatActivity)context;
+
 			if (backButtonHandler != null)
 			{
-				using (var icon = await context.GetFormsDrawable(backButtonHandler.IconOverride))
-				using (var mutatedIcon = icon.GetConstantState().NewDrawable().Mutate())
+				await UpdateDrawerArrowFromBackButtonBehavior(context, toolbar, drawerLayout, backButtonHandler, activity);
+			}
+			else
+			{
+				await UpdateDrawerArrow(context, toolbar, drawerLayout, activity);
+			}
+		}
+
+		protected virtual async Task UpdateDrawerArrow(Context context, Toolbar toolbar, DrawerLayout drawerLayout, FormsAppCompatActivity activity)
+		{
+			if (_drawerToggle == null)
+			{
+				_drawerToggle = new ActionBarDrawerToggle((Activity)context, drawerLayout, toolbar,
+					R.String.Ok, R.String.Ok)
 				{
-					mutatedIcon.SetColorFilter(TintColor.ToAndroid(Color.White), PorterDuff.Mode.SrcAtop);
-					toolbar.NavigationIcon = mutatedIcon;
+					ToolbarNavigationClickListener = this,
+				};
+
+				await UpdateDrawerArrowFromFlyoutIcon(context, _drawerToggle);
+
+				_drawerToggle.DrawerSlideAnimationEnabled = false;
+				drawerLayout.AddDrawerListener(_drawerToggle);
+			}
+
+			if (CanNavigateBack)
+			{
+				_drawerToggle.DrawerIndicatorEnabled = false;
+				using (var icon = new DrawerArrowDrawable(activity.SupportActionBar.ThemedContext))
+				{
+					icon.SetColorFilter(TintColor.ToAndroid(Color.White), PorterDuff.Mode.SrcAtop);
+					icon.Progress = 1;
+					toolbar.NavigationIcon = icon;
 				}
 			}
+			else if (_flyoutBehavior == FlyoutBehavior.Flyout)
+			{
+				toolbar.NavigationIcon = null;
+				var drawable = _drawerToggle.DrawerArrowDrawable;
+				drawable.SetColorFilter(TintColor.ToAndroid(Color.White), PorterDuff.Mode.SrcAtop);
+				_drawerToggle.DrawerIndicatorEnabled = true;
+			}
 			else
 			{
-				var activity = (FormsAppCompatActivity)context;
-				if (_drawerToggle == null)
+				_drawerToggle.DrawerIndicatorEnabled = false;
+			}
+			_drawerToggle.SyncState();
+		}
+
+		protected virtual async Task UpdateDrawerArrowFromBackButtonBehavior(Context context, Toolbar toolbar, DrawerLayout drawerLayout, BackButtonBehavior backButtonHandler, FormsAppCompatActivity activity)
+		{
+			var behavior = backButtonHandler;
+			var command = behavior.Command;
+			var commandParameter = behavior.CommandParameter;
+			var image = behavior.IconOverride;
+			var text = behavior.TextOverride;
+			var enabled = behavior.IsEnabled;
+
+			Drawable icon = null;
+
+			if (image != null)
+				icon = await context.GetFormsDrawable(image);
+
+			if (text != null)
+				icon = new FlyoutIconDrawerDrawable(context, TintColor, null, text);
```

Comment:
`TextOverride` has higher priority over `IconOverride`. I think they can be used at the same time.
```suggestion
				icon = new FlyoutIconDrawerDrawable(context, TintColor, icon, text);
```